feat: 2.12 Session Expiry modal cp-13.0.0

lionellbriones opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Vault Expiry info modal: This PR introduces a connections expiry modal for a rare edge case where the user is unable to unlock their local vault, and only global vault data can be recovered.

This situation occurs when a user logs in on a device using a very old password—older than the 20 most recent ones. In such cases, the user can still recover SRPs and keys, but local vault data—such as snaps, hardware connection IDs, and account counts—will not be retrievable.

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34118?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**
To test the password syncs, we can also use two browsers (with different profiles), if we don't have two devices.

1. Create (or Login) wallet on Device A (or in browser A)
2. Login using the same social account on Device B (or browser B with different Profile)
3. Lock the wallet on device B (Browser B).
2. Reset the wallet password 20+ times on the Device A (Browser A)
4. Unlock the wallet on the device B. After long loading period, you should see the modal. 
> This is to simulate the device B as an very old device which you has not use for a long time.
The long loading time is because we are trying to authenticate the very old password you have it in the device B.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**
![Screenshot 2025-07-08 at 12 32 23 PM](https://github.com/user-attachments/assets/3035bddc-ba8a-41e1-9b6d-e50c15b31838)



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [X] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [X] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

lionellbriones committed:
feat: added connections removed modal

lionellbriones committed:
Merge branch 'main' into feat/connections-removed

lwin-kyaw committed:
feat: update seedless onboarding controller and oauth getToken response

lwin-kyaw committed:
Merge remote-tracking branch 'origin/main' into feat/metadata-access-token

chaitanyapotti committed:
Merge branch 'main' into feat/connections-removed

lwin-kyaw committed:
feat: updated seedless-onboarding-controller to v2.1.0

lwin-kyaw committed:
Merge remote-tracking branch 'origin/main' into feat/metadata-access-token

lionellbriones committed:
Merge branch 'main' into feat/connections-removed

himanshuchawla009 committed:
wip: key chain max length

tuna1207 committed:
feat: rehydrate vault on pwd sync expired

lwin-kyaw committed:
fix: update refreshOAuthToken

metamaskbot committed:
Update LavaMoat policies

lionellbriones committed:
feat: add showConnectionsRemovedModal state

lwin-kyaw committed:
feat: updated SeedlessController network

tuna1207 committed:
feat: update controller

lionellbriones committed:
feat: integrate maxKeyChainLengthReached with UI

himanshuchawla009 commented:
I have read the CLA Document and I hereby sign the CLA

lionellbriones committed:
Merge branch 'main' into feat/connections-removed

himanshuchawla009 committed:
update js doc, add max key chain length

himanshuchawla009 committed:
Merge branch 'main' into feat/connections-removed

himanshuchawla009 committed:
policy json updated

himanshuchawla009 committed:
lint fixed

himanshuchawla009 committed:
policy json updated

metamaskbot committed:
Update LavaMoat policies

himanshuchawla009 committed:
Merge branch 'main' into feat/connections-removed

himanshuchawla009 committed:
fix tests

himanshuchawla009 committed:
fix sync pwd unit test

lionellbriones committed:
chore: initial e2e for showing connections removed modal

himanshuchawla009 committed:
fix ui unit test for key chain max length popup

himanshuchawla009 committed:
revert invalid e2e test

himanshuchawla009 committed:
fix connection popup show state

himanshuchawla009 committed:
update state snapshot

himanshuchawla009 committed:
Merge branch 'main' into feat/connections-removed

lionellbriones committed:
feat: remove learn more from connections removed modal body text

lwin-kyaw commented on lavamoat/browserify/beta/policy.json:
```diff
@@ -2194,8 +2194,7 @@
         "@noble/ciphers": true,
         "@noble/curves": true,
-        "@metamask/name-controller>async-mutex": true,
-        "browserify>buffer": true
```
Hi, `Buffer` usage was removed from the SeedlessOnboarding controller in `v2.3.0`.

cc: @MetaMask/policy-reviewers

lwin-kyaw committed:
feat: refresh token before create new vault for MaxKeyChainLengthExceeded case

lwin-kyaw committed:
test: add e2e tests for session-expiry

himanshuchawla009 committed:
Merge branch 'main' into feat/connections-removed

chaitanyapotti committed:
fix: add social login domains to blocked domains for provider injection

lwin-kyaw committed:
feat: update controller to 2.4.0

lwin-kyaw committed:
Merge branch 'main' into feat/connections-removed

lwin-kyaw committed:
chore: removed new Test Srp and reused the existing one.

lwin-kyaw commented on ui/components/app/connections-removed-modal/connections-removed-modal.stories.tsx:
```diff
@@ -0,0 +1,17 @@
+      <ConnectionsRemovedModal />
+    </>
+  );
+};
```
Simplified it here, https://github.com/MetaMask/metamask-extension/pull/34118/commits/e4fc4e49dacfff976c477841690ddef77d33159e

lwin-kyaw commented on test/e2e/helpers/seedless-onboarding/constants.ts:
```diff
@@ -1,3 +1,7 @@
+export const MockSeedPhrase =
+  'horse pupil style brush dust borrow simple easily bean margin actor valve';
```
Hi, removed the new Test SRP and reused the existing `E2E_SRP` here, https://github.com/MetaMask/metamask-extension/pull/34118/commits/e4fc4e49dacfff976c477841690ddef77d33159e

lwin-kyaw committed:
fix: fixed e2e

himanshuchawla009 committed:
fix: outdated pwd error cp-13.0.0 (#34441)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Fixes: #34445 
<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34441?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [X] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [X] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
